### PR TITLE
Delete inactive teams with no submissions.

### PR DESCRIPTION
**Pull request by apollo1291**
fixes Issue #363. deletes empty teams that have no submissions once the last user leaves. Does not delete teams that already have no users.

### CODE REVIEW

**Comment by gaurav-arya** [comment]
Should this still be a draft or marked ready for review?

**Review by gaurav-arya — CHANGES_REQUESTED** [review]
Looks good to me other than one nit!

File: backend/siarnaq/api/teams/signals.py
Comment (apollo1291) [inline backend/siarnaq/api/teams/signals.py:32]: should this function name change?

Comment (gaurav-arya) [reply backend/siarnaq/api/teams/signals.py:32]: Sure, how about `handle_empty_team`?

**Review by gaurav-arya — DISMISSED** [review]
Looks good to me! Let's get a review from @j-mao

File: backend/siarnaq/api/teams/signals.py
Comment (j-mao) [inline backend/siarnaq/api/teams/signals.py:35]: typo
```suggestion
            if not instance.has_active_submission():
```

**Review by j-mao — CHANGES_REQUESTED** [review]
This is almost good to go! Gave it a whirl and encountered a small bug that need fixing.

You'll likely also need to  change `TeamProfile.team` to have `on_delete=models.CASCADE` (instead of `PROTECT`). The reason is, "protect" will block deleting anything that has a profile attached, while "cascade" will delete any attached profiles. After this change, you might need to generate a new *migration* (this tells the database there's an update to the models): the command is `./manage.py makemigrations`, but feel free to give a yell in slack if something funky happens.

Once it's fixed and it's tested+working, feel free to send through another review! :')

**Review by gaurav-arya — APPROVED** [review]
Looks good to me! Wait for a re-review from @j-mao to merge. Also make sure to live test the latest version to ensure no silly typos:)

**Review by j-mao — APPROVED** [review]
Looks good to go!

**Pull request merged**
